index.js

```diff
@@ -804,6 +820,44 @@ function maxDepth (tree) {
   return tree.children.map(maxDepth).reduce((prev, next) => Math.max(prev, next), 0) + 1
 }
 
+function createAnimation (opts, render, done) {
+  var start = Date.now()
+  var animationFrame = null
+  var dt = 0
+  var ease = 0
+
+  function nextFrame () {
+    dt = (Date.now() - start) / opts.duration
+    ease = opts.ease(dt > 1 ? 1 : dt)
+    render(ease)
+
+    if (ease === 1) {
+      animationFrame = null
+      done()
+    } else {
+      animationFrame = requestAnimationFrame(nextFrame)
+    }
+  }
+  animationFrame = requestAnimationFrame(nextFrame)
+
+  return {
+    cancel () {
+      cancelAnimationFrame(animationFrame)
+    },
+    currentX (node) {
+      var prev = node.data.prev
+      return {
+        x0: interpolate(prev.x0, node.x0, ease),
+        x1: interpolate(prev.x1, node.x1, ease)
+      }
+    }
+  }
```

Comment:
I don't think we are transpiling in 0x, and this syntax won't work there.

Reply:
We aren't but 0x already extensively uses ES6 in its client side code. I didn't realise I was adding ES6 to an until now all-ES5 package so am happy to use ES5 instead, but it works as is in all recent browsers.

Reply:
Ok then we can land this and maybe reassess later!